simulator/sender.cc

```diff
@@ -238,49 +241,55 @@ void* Sender::MasterThread(int /*unused*/) {
         }
         // Add inter-symbol delay
         DelayTicks(tick_start, GetTicksForFrame(ctag.frame_id_) * symbol_delay);
-        tick_start = GetTime::Rdtsc();
+        tick_start += (GetTicksForFrame(ctag.frame_id_) * symbol_delay);
 
         size_t next_frame_id = ctag.frame_id_;
         // Check to see if the current frame is finished
         assert(next_symbol_id <= cfg_->Frame().NumTotalSyms());
         if (next_symbol_id == cfg_->Frame().NumTotalSyms()) {
+          // Set the end time to time the last symbol was scheduled
+          // to get entire frame length it will be necessary to add another
+          // symbol time or take start(frame+1) - start(frame)
+          frame_end_us = timestamp_us;
           next_frame_id++;
 
           // Find start symbol of next frame and add proper delay
           next_symbol_id = FindNextSymbol(0);
           if ((kDebugSenderReceiver == true) ||
               (kDebugPrintPerFrameDone == true)) {
-            double timeus_now = GetTime::GetTimeUs();
             std::printf(
                 "Sender: Tx frame %d in %.2f ms, next frame %zu, start symbol "
                 "%zu\n",
-                ctag.frame_id_, (timeus_now - start_time) / 1000.0,
+                ctag.frame_id_, (frame_end_us - frame_start_us) / 1000.0,
                 next_frame_id, next_symbol_id);
-            start_time = timeus_now;
           }
+          // Set end of frame time to the time after the last symbol
+          this->frame_end_[(ctag.frame_id_ % kNumStatsFrames)] = frame_end_us;
```

Comment:
We can simply use `GetTime::GetTimeUs()` to measure the timestamp here instead of using timestamp_us. We only need to measure the timestamp every frame instead of every symbol.

Reply:
If using GetTimeUs() here it would represent the completion time of the final symbol.  This might make sense but the start time is purely dependent on the schedule (main) thread where this change would make the finish time dependent on worker threads when have the bandwidth to transmit and complete the final symbol (and let the main thread know).